src/classes/Player.js

```diff
@@ -132,6 +124,42 @@ class Player {
     }
   }
 
+  async createStream() {
+    const url = queue[this.songEntry];
+    if (url.includes('youtube.com')) {
+      const stream = this.createYoutubeStream()
+      this.song = info.title;
+
+      stream.once(streamEvents.info, ({ title }) => {
+        this.song = title;
+        if (!this.updateDispatcherStatus()) {
+          this.updateSongPresence();
+        }
+      });
```

Comment:
Move this inside the `createYoutubeStream()` method and make sure you return the `stream` object. Also, remove `this.song = info.title;` because that line isn't doing anything there since `info` does not exist in that scope.